src/data/set/intervals/ord_connected.lean

```diff
@@ -106,3 +106,81 @@ lemma ord_connected_iff_interval_subset {s : set β} :
   λ h, ord_connected_iff.2 $ λ x hx y hy hxy, by simpa only [interval_of_le hxy] using h hx hy⟩
 
 end set
+
+section strict_mono
+variables {α : Type*} {β : Type*} [linear_order α] [preorder β] {f : α → β}
+
+-- these lemmas don't especially belong in this file, but they have the same imports, which is rare.
```

Comment:
I'd prefer to see these lemmas in another file (e.g., `data/set/intervals/surj_on` or `data/set/intervals/strict_mono`).